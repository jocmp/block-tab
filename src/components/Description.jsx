import React from 'react';
import ImageZoom from 'react-medium-image-zoom';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Description = ({ tab, active }) => {
  const { text, media } = tab;
  return (
    <div className={ classNames('description', { 'description--active': active }) }>
      <div className={ classNames('description__content', { 'description__content--active': active }) }>
        <div dangerouslySetInnerHTML={ { __html: text } } />
        { !!media &&
          <figure className="description__figure">
            <ImageZoom
              image={ {
                src: media,
                className: 'description__media'
              } }
            />
          </figure>
        }
      </div>
    </div>
  );
};

Description.propTypes = {
  active: PropTypes.bool.isRequired,
  tab: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    media: PropTypes.string
  })
};

export default Description;
