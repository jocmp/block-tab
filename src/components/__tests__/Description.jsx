import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Description from '../Description';

describe('Description', () => {
  const tab = () => (
    { id: '0101', title: 'Title 3', description: "<p>Description here</p>", media: "fake://img.con" }
  );

  it('renders correctly', () => {
    const wrapper = shallow(<Description tab={ tab() } active={ true } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('handles empty description', () => {
    const wrapper = shallow(<Description tab={ {} } active={ true } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders inactive state', () => {
    const wrapper = shallow(<Description tab={ {} } active={ false } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
