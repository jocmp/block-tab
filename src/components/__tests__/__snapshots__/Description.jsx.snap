// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description handles empty description 1`] = `
<div
  className="description description--active"
>
  <div
    className="description__content description__content--active"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Description renders correctly 1`] = `
<div
  className="description description--active"
>
  <div
    className="description__content description__content--active"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <figure
      className="description__figure"
    >
      <ImageZoom
        defaultStyles={
          Object {
            "image": Object {},
            "overlay": Object {},
            "zoomContainer": Object {},
            "zoomImage": Object {},
          }
        }
        image={
          Object {
            "className": "description__media",
            "src": "fake://img.con",
          }
        }
        onUnzoom={[Function]}
        onZoom={[Function]}
        shouldHandleZoom={[Function]}
        shouldReplaceImage={true}
        shouldRespectMaxDimension={false}
        zoomMargin={40}
      />
    </figure>
  </div>
</div>
`;

exports[`Description renders inactive state 1`] = `
<div
  className="description"
>
  <div
    className="description__content"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;
